tests/integration/rApi.spec.ts

```diff
@@ -125,55 +155,140 @@ describe("endpoints which consume the R API", { sequential: true }, async () =>
   // to the R API. This is called a 'rule' in Mockoon, and rules can't be used simultaneously with the 'sequential' setting, so we
   // instead use the mockoonResponse parameter to tell Mockoon which type of response to send.
   describe("post api/scenarios", async () => {
```

Comment:
From this point, the changes in this file are basically:

* Nest the existing three tests of `post api/scenarios` under a describe block `"when a scenario with these parameters has not been run before".
* Update the first of those tests so that it also checks for the scenario record having been created correctly.
* Add two other sibling describe blocks: `"when a scenario with these parameters has already been run before"` and `""when a scenario with these parameters has already been run before, but the R API does not recognise the run id"`, each with a single test inside.

Reply:
* Refactor the requests to /api/scenarios into a convenient requestNewScenario function
* Add a test for when the /api/versions request fails